contracts/0.4.24/Lido.sol

```diff
@@ -7,255 +7,326 @@ pragma solidity 0.4.24;
+    // "beacon" in the `keccak256()` parameter is staying here for compatibility reason
+    bytes32 internal constant CL_BALANCE_POSITION = keccak256("lido.Lido.beaconBalance");
+    /// @dev number of Lido's validators available in the Consensus Layer state
+    // "beacon" in the `keccak256()` parameter is staying here for compatibility reason
+    bytes32 internal constant CL_VALIDATORS_POSITION = keccak256("lido.Lido.beaconValidators");
+    /// @dev positive token rebase allowed per single LidoOracle report
+    /// uses 1e9 precision, e.g.: 1e6 - 0.1%; 1e9 - 100%, see `setMaxPositiveTokenRebase()`
+    bytes32 internal constant MAX_POSITIVE_TOKEN_REBASE_POSITION = keccak256("lido.Lido.MaxPositiveTokenRebase");
+    /// @dev Just a counter of total amount of execution layer rewards received by Lido contract. Not used in the logic.
+    bytes32 internal constant TOTAL_EL_REWARDS_COLLECTED_POSITION = keccak256("lido.Lido.totalELRewardsCollected");
+    event Stopped();
+    event Resumed();
+
+    event StakingPaused();
+    event StakingResumed();
+    event StakingLimitSet(uint256 maxStakeLimit, uint256 stakeLimitIncreasePerBlock);
+    event StakingLimitRemoved();
+
+    event ETHDistributed(
+        int256 clBalanceDiff,
+        uint256 withdrawalsWithdrawn,
+        uint256 executionLayerRewardsWithdrawn,
+        uint256 postBufferredEther
+    );
 
-    /// @dev percent in basis points of total pooled ether allowed to withdraw from LidoExecutionLayerRewardsVault per LidoOracle report
-    bytes32 internal constant EL_REWARDS_WITHDRAWAL_LIMIT_POSITION = keccak256("lido.Lido.ELRewardsWithdrawalLimit");
+    event TokenRebase(
+        uint256 preTotalShares,
+        uint256 preTotalEther,
+        uint256 postTotalShares,
+        uint256 postTotalEther,
+        uint256 sharesMintedAsFees,
+        uint256 timeElapsed
+    );
 
-    /// @dev Just a counter of total amount of execution layer rewards received by Lido contract
-    /// Not used in the logic
-    bytes32 internal constant TOTAL_EL_REWARDS_COLLECTED_POSITION = keccak256("lido.Lido.totalELRewardsCollected");
+    // Lido locator set
+    event LidoLocatorSet(address lidoLocator);
+
+    // The amount of ETH withdrawn from LidoExecutionLayerRewardsVault to Lido
+    event ELRewardsReceived(uint256 amount);
 
-    /// @dev Credentials which allows the DAO to withdraw Ether on the 2.0 side
-    bytes32 internal constant WITHDRAWAL_CREDENTIALS_POSITION = keccak256("lido.Lido.withdrawalCredentials");
+    // The amount of ETH withdrawn from WithdrawalVault to Lido
+    event WithdrawalsReceived(uint256 amount);
+
+    // Max positive token rebase set (see `setMaxPositiveTokenRebase()`)
+    event MaxPositiveTokenRebaseSet(uint256 maxPositiveTokenRebase);
+
+    // Records a deposit made by a user
+    event Submitted(address indexed sender, uint256 amount, address referral);
+
+    // The `amount` of ether was sent to the deposit_contract.deposit function
```

Comment:
// The `amount` of ether was sent to the staking router

???

Reply:
I dont' know - to scary - maybe maybe maybe

Reply:
Someone would use this event as signal that this ether has been sent to deposit contract.